src/wp-admin/includes/ms.php

```diff
@@ -98,7 +98,7 @@ function wpmu_delete_blog( $blog_id, $drop = false ) {
 	}
 
 	// Don't destroy the initial, main, or root blog.
-	if ( $drop && ( 1 == $blog_id || is_main_site( $blog_id ) || ( $blog->path == $current_network->path && $blog->domain == $current_network->domain ) ) ) {
+	if ( $drop && $blog && ( 1 == $blog_id || is_main_site( $blog_id ) || ( $blog->path == $current_network->path && $blog->domain == $current_network->domain ) ) ) {
```

Comment:
Why isn't this being caught in the `if ( $drop && ! $blog )` check just before it?

Reply:
Hmm, that's a good question. The issue this was solving was that $blog wasn't an object. 

See: https://travis-ci.org/WordPress/wordpress-develop/jobs/445966625#L1411

I think that there is a chance we aren't setting the environment up completely correctly and that could be causing this.